### ISSUE TITLE
1558M support

### ISSUE THREAD

**Issue by minimaxir** [minimaxir/gpt-2-simple#136]
Verify that the largest model is supported:

* Finetuning is disabled
* Generating from default is supported
* Update docs.

**Comment by minimaxir** [comment]
Testing in the Notebook, the 1558M model does not fit in a K80. Will check letter to see if I get lucky and get a T4/P100.

Edited copy in the notebook to address 1558M model.

**Comment by lendrick** [comment]
Thought I'd comment on this, since I've gotten it working:

I've got a fairly beefy machine with 32 gigs of ram.  As long as I use tensorflow and not tensorflow_gpu, the 1558M model works for me.  RAM usage spikes at ~11 gigs, so I don't think anyone will have any luck with any consumer-grade GPUs, but they might be able to squeeze it into 16 gigs if they aren't running Chrome. :)

**Comment by WAUthethird** [comment]
> Thought I'd comment on this, since I've gotten it working:
> 
> I've got a fairly beefy machine with 32 gigs of ram. As long as I use tensorflow and not tensorflow_gpu, the 1558M model works for me. RAM usage spikes at ~11 gigs, so I don't think anyone will have any luck with any consumer-grade GPUs, but they might be able to squeeze it into 16 gigs if they aren't running Chrome. :)

Are you training or generating? If generating, what parameters are you using? I have 16 gigs installed, but I can't seem to get the full sample size to generate without getting an OOM error.

**Comment by TGower** [comment]
I've been able to generate samples with the 1558M model on GPU inside a 1080ti. Memory usage is 10652MiB / 11178MiB. Driving gpt-2-simple via gpt2_client script. I couldn't get it to run on the gpu when I installed gpt_client from pip, but copying files from https://github.com/rish-16/gpt2client/tree/master/gpt2_client and running on a fresh conda tensorflow-gpu=1.14.0 environment cleared everything up for me.

**Comment by gpt2ent** [comment]
> Testing in the Notebook, the 1558M model does not fit in a K80. Will check letter to see if I get lucky and get a T4/P100.
Fits only in a P100.

**Comment by WAUthethird** [comment]
Works in a P100, please remove the inability to finetune.
At the very most, use a warning.

**Comment by backnotprop** [comment]
Ive finetuned some models that end up around 5gb.  The generation process is super slow (minutes ...  about 40% loading & 60% generating wait time)

How can I speed up?

**Comment by minimaxir** [comment]
Agree with depreciating to a warning since some have successfully got it working.

**Comment by GoAwayNow** [comment]
Been able to generate using 1558M on P100 for sure, OOM on finetuning still, though.

**Comment by Grandiferr** [comment]
I'm a beginner and I'm really confused. Could someone please explain to me this. Can I finetune the 1558M model in Google Colab? If yes, how. If not, could someone point me to a tutorial on setting up this thing on P100?

Because I've been consuming more info than I'm able to comprehend lately.

**Comment by minimaxir** [comment]
tl;dr no, you cannot finetune 1558M in Colab.

**Comment by Grandiferr** [comment]
What do I need then? What setup is required?

Sorry if I'm being too dumb. I just really want to figure it out :D

**Comment by sophiawisdom** [comment]
I can't finetune a 1558M on an AWS g4 == NVIDIA t4, i consistently OOM.
